### PR TITLE
SK-1619: DID Upgrade option

### PR DESCRIPTION
**Pull request by lukeg90**
SK-1619

### CODE REVIEW

File: src/components/IdentityOptions/IdentityOptions.tsx [hunk 1]
```diff
@@ -17,9 +18,12 @@ interface Props {
 export function IdentityOptions({ identity, onEdit }: Props): JSX.Element {
   const t = browser.i18n.getMessage;
 
-  const { address } = identity;
+  const { address, did } = identity;
 
-  const { buttonProps, itemProps, isOpen, setIsOpen } = useDropdownMenu(3);
+  const isLightDid = Boolean(did) && !isFullDid(did);
```
Comment (arty-name) [inline src/components/IdentityOptions/IdentityOptions.tsx:23]: Should it also check whether the DID has been deleted?

File: src/utilities/useTXDSubmitter/useTXDSubmitter.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  'wss://sporran-testnet.kilt.io': 'https://txd-dev.trusted-entity.io',
+};
+
+async function getTXDSubmitter() {
+  const endpoint = await getEndpoint();
+  const TXD = TXDURLs[endpoint];
+
+  const { paymentAddress } = await ky
+    .get(`${TXD}/meta`)
+    .json<{ paymentAddress: string }>();
```
Comment (arty-name) [inline src/utilities/useTXDSubmitter/useTXDSubmitter.ts:24]: Would this work even nicer? 
```suggestion
    .json<{ paymentAddress: KiltAddress }>();
```

File: src/views/DidUpgrade/DidUpgrade.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        document,
+        keypair,
+        sign,
+        submitter,
+      );
+
+      const checkout = await getCheckoutURL();
+
+      const url = new URL(checkout);
+      url.searchParams.set('address', address);
```
Comment (arty-name) [inline src/views/DidUpgrade/DidUpgrade.tsx:66]: While the checkout does not support it yet, we want to have it in the future, so we can use this moment to include `url.searchParams.set('did', fullDid);` here

**Review by arty-name — APPROVED** [review]

File: src/components/IdentityOptions/IdentityOptions.tsx (hunk 1 above)
Comment (lukeg90) [reply src/components/IdentityOptions/IdentityOptions.tsx:23]: Isn't that already checked by `syncDidStateWithBlockchain` which sets did to undefined if it was deleted?

Comment (arty-name) [reply src/components/IdentityOptions/IdentityOptions.tsx:23]: Sorry, I misread the code, all good :+1:

**Pull request merged**